drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -346,7 +346,6 @@ end
 local function find_default_endpoint(device, component)
   local switch_eps = device:get_endpoints(clusters.OnOff.ID)
   local button_eps = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.Feature.MOMENTARY_SWITCH})
-  local valve_eps = embedded_cluster_utils.get_endpoints(device, clusters.ValveConfigurationAndControl.ID)
```

Comment:
Why is this removed?

Reply:
I think that we don't need it here because water valve was moved from being initialized in `initialize_switch` to `do_configure`, like electrical sensor. `find_default_endpoint` is really only used in `initialize_switch` for device types that we expect to contain multiple endpoints, and therefore they don't need any parent-child or multi component configuration, and simply need to be mapped to a profile. So now `initialize_switch` is only for switch and button devices and `do_configure` handles the rest. Do you agree with this implementation? Would it make more sense to handle them all in `initialize_switch`?

Reply:
do_configure will be called before device_init (which is where initialize switch is called) so I think it is best to keep the devices that don't use the initialize_switch functionality separated from there. It's syntactical sugar in this case but useful for keeping things organized I think.